composer/trainer/trainer_hparams.py

```diff
@@ -177,8 +177,34 @@ class TrainerHparams(hp.Hparams):
         default=-1)
     callbacks: List[CallbackHparams] = hp.optional(doc="Callback hparams", default_factory=list)
 
-    load_checkpoint: Optional[CheckpointLoaderHparams] = hp.optional(doc="Checkpoint loading hparams", default=None)
-    save_checkpoint: Optional[CheckpointSaverHparams] = hp.optional(doc="Checkpointing hparams", default=None)
+    load_path: Optional[str] = hp.optional(doc=textwrap.dedent("""If specified, the path to an existing checkpoint file
+        (if the checkpoint is on the local disk) or the object name for the checkpoint
+        (if the checkpoint is in a cloud bucket). Set to None (the default) to skip loading from a checkpoint."""),
+                                           default=None)
+    load_object_store: Optional[ObjectStoreProviderHparams] = hp.optional(doc=textwrap.dedent("""
+        If the checkpoint is in an object store (i.e. AWS S3 or Google Cloud Storage), the parameters for
+        connecting to the cloud provider object store. Otherwise, if the checkpoint is a local filepath,
+        leave blank."""),
+                                                                          default=None)
+    load_weights_only: bool = hp.optional(doc="Whether to only load the weights from the model.", default=False)
+    load_strict_model_weights: bool = hp.optional(
+        doc="Ensure that the set of wcheckpoint_eights in the checkpoint and model must exactly match.", default=False)
+
+    load_chunk_size: int = hp.optional(doc=textwrap.dedent("""Chunk size (in bytes) to use when downloading checkpoints.
+        Ignored if the checkpoint is a local file path."""),
+                                       default=1_048_576)
+    load_progress_bar: bool = hp.optional(
+        doc=textwrap.dedent("""Whether to show a progress bar when downloading checkpoints.
+        Ignored if the checkpoint is a local file path"""),
+        default=True)
+
+    save_folder: Optional[str] = hp.optional(doc=textwrap.dedent(f"""If set, folder to save checkpoint files.
+        Relative to the run directory. Defaults to None, meaning checkpoints will not be saved."""),
+                                             default=None)
+    save_interval: str = hp.optional(doc=textwrap.dedent("""The time string interval representing how
+        frequently checkpoints should be saved. For example, set to "1ep" to save checkpoints every epoch, or "10ba"
+        to save checkpoints every 10 batches.. """),
+                                     default="1ep")
```

Comment:
The thing I don't like about this is as a user, it's now unclear what the difference in configuration is between a config that loads a checkpoint and a config that does not load a checkpoint. "Do you have to specify all of the `load_*` hparams if you want to turn on checkpoint loading?" is a question I imagine many users would have.

Since we're committed to flattening, could we add a `load_checkpoint: Optional[bool]` field? If `load_checkpoint` is true, we attempt load a checkpoint, otherwise we don't. In `.validate()` for the `TrainerHparams`, we would then also validate that `load_path` is required if `load_checkpoint` is true, and that all of the `load_*` must be null if `load_checkpoint` is false.

And likewise for `save_checkpoint`.

Reply:
Would it be natural for users to think that save_folder and load_path serve these purposes?

Reply:
@ravi-mosaicml and I just discussed offline and we think, since we're committed to flattening configs, we're indeed best off substituting static typing for copious documentation. Probably not going to add unnecessary fields to the hparams.

Reply:
Added comments to the docstrings to make it clear that secondary parameters have no effect unless if the paths are set.